core/src/main/java/com/spotify/ffwd/AgentConfig.kt

```diff
@@ -16,53 +16,83 @@
 
 package com.spotify.ffwd
 
-import com.fasterxml.jackson.annotation.JsonCreator
-import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.module.SimpleModule
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module
 import com.spotify.ffwd.domain.SearchDomainDiscovery
 import com.spotify.ffwd.input.InputManagerModule
 import com.spotify.ffwd.output.OutputManagerModule
+import com.uchuhimo.konf.Config
+import com.uchuhimo.konf.ConfigSpec
 import java.net.InetAddress
 import java.net.InetSocketAddress
 import java.net.UnknownHostException
 import java.nio.file.Path
-import java.nio.file.Paths
 
-data class AgentConfig(
-    @JsonProperty("debug") var debug: Debug?,
-    @JsonProperty("host") var host: String = buildDefaultHost(),
-    @JsonProperty("tags") var tags: Map<String, String> = emptyMap(),
-    @JsonProperty("tagsToResource") var tagsToResource: Map<String, String> = emptyMap(),
-    @JsonProperty("riemannTags") var riemannTags: Set<String> = emptySet(),
-    @JsonProperty("skipTagsForKeys") var skipTagsForKeys: Set<String> = emptySet(),
-    @JsonProperty("automaticHostTag") var automaticHostTag: Boolean = true,
-    var input: InputManagerModule =
-        InputManagerModule.supplyDefault().get(),
-    @JsonProperty("output") var output: OutputManagerModule =
-        OutputManagerModule.supplyDefault().get(),
-    @JsonProperty("searchDomain") var searchDomain: SearchDomainDiscovery =
-        SearchDomainDiscovery.supplyDefault(),
-    @JsonProperty("asyncThreads") var asyncThreads: Int = 4,
-    @JsonProperty("schedulerThreads") var schedulerThreads: Int = 4,
-    @JsonProperty("bossThreads") var bossThreads: Int = 2,
-    @JsonProperty("workerThreads") var workerThreads: Int = 4,
-    @JsonProperty("ttl") var ttl: Long = 0,
-    // NB(hexedpackets): qlog is unused and can be removed once the config parser ignores unknown
-    // properties.
-    @JsonProperty("qlog") var qlog: String?
-)
+// Helper class to make interop with java easier. The configuration loading is done through the
+// static object.
+class AgentConfig(val config: Config) {
+    fun hasDebug(): Boolean = config.contains(Debug.host) or config.contains(Debug.port)
+
+    val debugLocalAddress =  config[Debug.localAddress]
+    val host = config[AgentConfig.host]
+    val tags = config[AgentConfig.tags]
+    val tagsToResource = config[AgentConfig.tagsToResource]
+    val riemannTags = config[AgentConfig.riemannTags]
+    val skipTagsForKeys = config[AgentConfig.skipTagsForKeys]
+    val automaticHostTag = config[AgentConfig.automaticHostTag]
+    val input: InputManagerModule = config[AgentConfig.input]
+    val output: OutputManagerModule = config[AgentConfig.output]
+    val searchDomain = config[AgentConfig.searchDomain]
```

Comment:
should this be SearchDomainDiscovery?

Reply:
The type is `SearchDomainDiscovery` but I kept the original variable name of `searchDomain`.

Reply:
the object type still is - down below the config will return ` val searchDomain by lazy { SearchDomainDiscovery.supplyDefault() }`

Reply:
👍